### PR TITLE
[Tests] Add tests to the Component class

### PR DESCRIPTION
**Pull request by Flixow**

### CODE REVIEW

File: classes/Component.spec.js [hunk 1]
```diff
@@ -1,9 +1,269 @@
...
+        component.writeComponentStructure(template);
+
+        expect(writeComponentFileStub).to.have.been.calledWith(template);
+        expect(writeComponentIndexFileStub).to.have.been.called;
+        expect(manageComponentsIndexFileStub).to.have.been.called;
+      });
+
+      it('without component file if already exists', () => {
+        mock({
+          [`/${componentName}/${componentName}.js`]: Buffer.from([8, 6, 7, 5, 3, 0, 9]),
```
Comment (gh3r) [inline classes/Component.spec.js]: why you use Buffer here and why this array of numbers?

File: package.json [hunk 2]
```diff
@@ -20,8 +20,10 @@
     "commander": "^2.18.0",
     "fs-extra": "^7.0.0",
     "mocha": "^5.2.0",
+    "mock-fs": "^4.7.0",
     "replace": "^1.0.0",
-    "sinon": "^6.3.4"
+    "sinon": "^6.3.4",
+    "sinon-chai": "^3.2.0"
```
Comment (gh3r) [inline package.json]: shouldn't we use exact versions of packages?

File: classes/Component.spec.js (hunk 1 above)
Comment (Flixow) [reply classes/Component.spec.js]: It's just a mock for a artificial file.

File: classes/Component.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ const { capitalize } = require('../utils');
 class Component {
   constructor(component, options) {
     this.component = capitalize(component);
-    this.options = options;
+    this.options = options || {};
```
Comment (hwesol13) [inline classes/Component.js]: If `options` param has value or is `undefined` (not null) you can use default value just in params line
```
constructor(component, options = {}) {
...
}
```

**Pull request closed without merging**